src/main/java/com/stripe/model/Customer.java

```diff
@@ -1,263 +1,335 @@
    */
-  public static Customer retrieve(String id) throws StripeException {
-    return retrieve(id, (RequestOptions) null);
+  public Customer delete(RequestOptions options) throws StripeException {
+    return delete((Map<String, Object>) null, options);
   }
 
   /**
-   * Retrieve a customer.
+   * Permanently deletes a customer. It cannot be undone. Also immediately cancels any active
+   * subscriptions on the customer.
    */
-  public static Customer retrieve(String id, RequestOptions options) throws StripeException {
-    return retrieve(id, null, options);
+  public Customer delete(Map<String, Object> params) throws StripeException {
+    return delete(params, (RequestOptions) null);
   }
 
   /**
-   * Retrieve a customer.
+   * Permanently deletes a customer. It cannot be undone. Also immediately cancels any active
+   * subscriptions on the customer.
    */
-  public static Customer retrieve(String id, Map<String, Object> params, RequestOptions options)
+  public Customer delete(Map<String, Object> params, RequestOptions options)
       throws StripeException {
-    return request(RequestMethod.GET, instanceUrl(Customer.class, id), params, Customer.class,
-        options);
+    String url =
+        String.format("%s%s", Stripe.getApiBase(), String.format("/v1/customers/%s", this.getId()));
+    return request(ApiResource.RequestMethod.DELETE, url, params, Customer.class, options);
   }
-  // </editor-fold>
 
-  // <editor-fold desc="update">
-  /**
-   * Update a customer.
-   */
-  @Override
-  public Customer update(Map<String, Object> params) throws StripeException {
-    return update(params, (RequestOptions) null);
+  /** Removes the currently applied discount on a customer. */
+  public Discount deleteDiscount() throws StripeException {
+    return deleteDiscount((Map<String, Object>) null, (RequestOptions) null);
   }
 
-  /**
-   * Update a customer.
-   */
-  @Override
-  public Customer update(Map<String, Object> params, RequestOptions options)
-      throws StripeException {
-    return request(RequestMethod.POST, instanceUrl(Customer.class, this.id), params,
-        Customer.class, options);
+  /** Removes the currently applied discount on a customer. */
+  public Discount deleteDiscount(Map<String, Object> params) throws StripeException {
+    return deleteDiscount(params, (RequestOptions) null);
   }
-  // </editor-fold>
 
-  @Getter
-  @Setter
-  @EqualsAndHashCode(callSuper = false)
-  public static class NextRecurringCharge extends StripeObject {
-    Long amount;
-    String date;
+  /** Removes the currently applied discount on a customer. */
+  public Discount deleteDiscount(Map<String, Object> params, RequestOptions options)
+      throws StripeException {
+    String url =
+        String.format(
+            "%s%s", Stripe.getApiBase(), String.format("/v1/customers/%s/discount", this.getId()));
+    return request(ApiResource.RequestMethod.DELETE, url, params, Discount.class, options);
```

Comment:
17) Delete (custom): Instance method to delete an arbitrary resource relying ID attribute of the instance.
Method name makes it distinct from standard delete method.

Reply:
https://github.com/stripe/stripe-java/blob/2aa5db5c39dc6e91228507798083f373904cc8c0/src/main/java/com/stripe/model/Customer.java#L285-L292